lib/smart_proxy_reports/friendly_message.rb

```diff
@@ -27,9 +27,15 @@ def generate_message
     msg
   end
 
+  def readable_tree(tree)
+    return tree if tree.nil?
+    return tree.slice(0, 10).join(", ") + " and more" if tree.length > 10
+    tree.join(", ")
+  end
```

Comment:
Could you consider a different name for this method? It accepts an array (or anything that responds to slice), probably should be called `readable_array`. Also consider making `10` an optional argument that can be provided on demand.